File: CS/GridView.UnboundColumns/Controllers/GridViewController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using GridView.UnboundColumns.Models;

namespace GridView.UnboundColumns.Controllers {
    [HandleError]
    public class GridViewController : Controller {
        public ActionResult Index() {
            return UnboundColumns();
        }
        public ActionResult UnboundColumns() {
            return View("UnboundColumns", NorthwindDataProvider.GetInvoices());
        }
        public ActionResult UnboundColumnsPartial() {
            return PartialView("UnboundColumnsPartial", NorthwindDataProvider.GetInvoices());
        }
    }
}

File: VB/GridView.UnboundColumns/Controllers/GridViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports GridView.UnboundColumns.Models

Namespace GridView.UnboundColumns.Controllers
	<HandleError> _
	Public Class GridViewController
		Inherits Controller
		Public Function Index() As ActionResult
			Return UnboundColumns()
		End Function
		Public Function UnboundColumns() As ActionResult
			Return View("UnboundColumns", NorthwindDataProvider.GetInvoices())
		End Function
		Public Function UnboundColumnsPartial() As ActionResult
			Return PartialView("UnboundColumnsPartial", NorthwindDataProvider.GetInvoices())
		End Function
	End Class
End Namespace
